hack/configapi/cmd/root.go

```diff
@@ -61,33 +64,73 @@ func runCmd(cmd *cobra.Command, _ []string) error {
 		Bucket: awsBucket,
 		Region: awsRegion,
 	}
-	privateKey, err := os.ReadFile(privateKeyPath)
-	if err != nil {
-		return fmt.Errorf("reading private key: %w", err)
-	}
-
 	versionBytes, err := os.ReadFile(versionFilePath)
 	if err != nil {
 		return fmt.Errorf("reading version file: %w", err)
 	}
-	var versions configapi.AzureSEVSNPVersion
-	err = json.Unmarshal(versionBytes, &versions)
+	var inputVersion attestationconfig.AzureSEVSNPVersion
+	err = json.Unmarshal(versionBytes, &inputVersion)
 	if err != nil {
 		return fmt.Errorf("unmarshalling version file: %w", err)
 	}
+	fetcher := fetcher.New()
+	latestAPIVersion, err := fetcher.FetchAzureSEVSNPVersionLatest(ctx)
+	if err != nil {
+		return fmt.Errorf("fetching latest version: %w", err)
+	}
 
-	sut, err := attestationconfigclient.New(ctx, cfg, []byte(cosignPwd), privateKey)
+	isNewer, err := IsInputNewerThanLatestAPI(inputVersion, latestAPIVersion.AzureSEVSNPVersion)
 	if err != nil {
-		return fmt.Errorf("creating repo: %w", err)
+		return fmt.Errorf("comparing versions: %w", err)
 	}
+	if isNewer {
+		privateKey, err := os.ReadFile(privateKeyPath)
+		if err != nil {
+			return fmt.Errorf("reading private key: %w", err)
+		}
+		sut, err := attestationconfigclient.New(ctx, cfg, []byte(cosignPwd), privateKey)
+		if err != nil {
+			return fmt.Errorf("creating repo: %w", err)
+		}
 
-	if err := sut.UploadAzureSEVSNP(ctx, versions, time.Now()); err != nil {
-		return fmt.Errorf("uploading version: %w", err)
+		if err := sut.UploadAzureSEVSNP(ctx, inputVersion, time.Now()); err != nil {
+			return fmt.Errorf("uploading version: %w", err)
+		}
+		cmd.Printf("Successfully uploaded new Azure SEV-SNP version: %+v\n", inputVersion)
+	} else {
+		cmd.Printf("Input version: %+v is not newer than latest API version: %+v\n", inputVersion, latestAPIVersion)
 	}
-	cmd.Printf("Successfully uploaded new Azure SEV-SNP version: %+v\n", versions)
 	return nil
 }
 
+// IsInputNewerThanLatestAPI compares all version fields with the latest API version and returns true if any input field is newer.
+func IsInputNewerThanLatestAPI(input, latest attestationconfig.AzureSEVSNPVersion) (bool, error) {
```

Comment:
Not entirely sure what is happening here :D, but I offer you an alternative that seems simpler to me and needs no reflections:
- Implement a method on `attestationconfig.AzureSEVSNPVersion` that is analogous to [semver.Compare](https://pkg.go.dev/golang.org/x/mod/semver#Compare). v is the object, w the argument. v is smaller than w if at least one version is smaller, v is larger than w if all versions of w are equal to v and at least one version of w is larger than w. 
- Call `input.Compare(latest)` here, sth like: `return input.Compare(latest) >= 0`.

maybe I am missing something..